### DISCUSSION TITLE
Links are connecting and reconnecting *datalinks* after changing from udp to tcp

### DISCUSSION THREAD

**Discussion by JohnAdam-123** [yamcs/yamcs#958] [General]
What might be the problem with the _datalinks_, after changing the configuration from _udp_ to _tcp_ in _yamcs.[instance].yaml_ I have been able to connect to the port and the interface that I wanted, but the connection is not stable. And this is also even when connected to localhost.

_18:19:32.588 myproject [30] TcpTcDataLink [tcp-tc-out] Cannot connect to 10.8.0.3:2002 'Connection refused'. Retrying in 10s
18:19:42.726 myproject [29] TcpTmDataLink [tcp-tm-in] Cannot open or read TM socket 10.8.0.3: 2001: Connection refused. Retrying in 10 seconds.
18:19:43.041 myproject [30] TcpTcDataLink [tcp-tc-out] Cannot connect to 10.8.0.3:2002 'Connection refused'. Retrying in 10s
18:19:53.769 myproject [29] TcpTmDataLink [tcp-tm-in] Cannot open or read TM socket 10.8.0.3: 2001: java.net.SocketTimeoutException: Connect timed out. Retrying in 10 seconds.
18:20:04.977 myproject [29] TcpTmDataLink [tcp-tm-in] Cannot open or read TM socket 10.8.0.3: 2001: java.net.SocketTimeoutException: Connect timed out. Retrying in 10 seconds.
07:06:32.838 myproject [29] TcpTmDataLink [tcp-tm-in] Cannot open or read TM socket 10.8.0.3: 2001: java.net.SocketTimeoutException: Connect timed out. Retrying in 10 seconds.
07:06:44.979 myproject [29] TcpTmDataLink [tcp-tm-in] Cannot open or read TM socket 10.8.0.3: 2001: java.net.SocketTimeoutException: Connect timed out. Retrying in 10 seconds.
07:06:56.067 myproject [29] TcpTmDataLink [tcp-tm-in] Cannot open or read TM socket 10.8.0.3: 2001: java.net.SocketTimeoutException: Connect timed out. Retrying in 10 seconds.
07:07:09.253 myproject [29] TcpTmDataLink [tcp-tm-in] Link established to 10.8.0.3:2001
07:07:15.914 myproject [30] TcpTcDataLink [tcp-tc-out] Link established to 10.8.0.3:2002_

**Comment by xpromache** [comment]
my guess is a windows antivirus or firewall acting up

**Reply by JohnAdam-123** [reply]
The above output is running in ubuntu 22.04, so I just wanted to know what might be the cause. 

okay... so it might be firewall configuration that's disturbs the connection... right.
